api/v1beta1/azurecluster_webhook.go

```diff
@@ -75,6 +75,20 @@ func (c *AzureCluster) ValidateUpdate(oldRaw runtime.Object) error {
 		)
 	}
 
+	if !reflect.DeepEqual(c.Spec.ControlPlaneEndpoint.Host, old.Spec.ControlPlaneEndpoint.Host) {
```

Comment:
this wouldn't allow me to go from `""` to `"hostname"` because "" != hostname. This should be 

```suggestion
	if !old.Spec.ControlPlaneEndpoint.Host != "" && !reflect.DeepEqual(c.Spec.ControlPlaneEndpoint.Host, old.Spec.ControlPlaneEndpoint.Host) {
```

Reply:
So, is there something that `reflect.DeepEqual()` gives us when we are just comparing two strings? Host isn't that complex is it?

Reply:
And this is more of a style question...

Which do you prefer?

```
if a != 'foo' && a != b { ... }
```
or
```
if !(a == 'foo'  || a == b) { ... }
```

I personally like the latter because I my brain turns the 'if !' into 'unless'. Then I read the conditionals as 'if a equals foo or a equals b skip the block'. However, I'm mostly seeing the former in the code, is that a go'ism?

Reply:
> So, is there something that reflect.DeepEqual() gives us when we are just comparing two strings?

not AFAIK. You can use `==`, should be the same.

> However, I'm mostly seeing the former in the code, is that a go'ism?

I don't think one is preferred over the other, it's mostly style / personal preference. Personally, I read the first as "if old is not empty and old is not equal to new, then return an error" which matches the intent of the validation here but logically they are exactly the same.